algebra-core/src/fields/mod.rs

```diff
@@ -191,50 +213,99 @@ pub trait FpParameters: 'static + Send + Sync + Sized {
     /// (Should equal `SELF::MODULUS_BITS - 1`)
     const CAPACITY: u32;
 
-    /// 2^s * t = MODULUS - 1 with t odd. This is the two-adicity of
-    /// `Self::MODULUS`.
-    const TWO_ADICITY: u32;
-
     /// t for 2^s * t = MODULUS - 1
     const T: Self::BigInt;
 
-    /// 2^s root of unity computed by GENERATOR^t
-    const ROOT_OF_UNITY: Self::BigInt;
-
     /// (t - 1) / 2
     const T_MINUS_ONE_DIV_TWO: Self::BigInt;
 
     /// (Self::MODULUS - 1) / 2
     const MODULUS_MINUS_ONE_DIV_TWO: Self::BigInt;
 }
 
+/// The interface for fields that are able to be used in FFTs.
+pub trait FftField: Field + From<u128> + From<u64> + From<u32> + From<u16> + From<u8> {
+    type FftParams: FftParameters;
+
+    /// Returns the 2^s root of unity.
+    fn root_of_unity() -> Self;
+
+    /// Returns the 2^s * small_subgroup_base^small_subgroup_power root of unity
+    /// if a small subgroup is defined.
+    fn full_root_of_unity() -> Option<Self>;
+
+    /// Returns the multiplicative generator of `char()` - 1 order.
+    fn multiplicative_generator() -> Self;
+
+    /// Returns the root of unity of order n (for n a power of 2), if one
+    /// exists.
```

Comment:
This comment isn't accurate any more, right?

Reply:
Yes, at least if there is a small subgroup defined it has a different semantic. I'll update the comment.